python/lsst/ctrl/mpexec/singleQuantumExecutor.py

```diff
@@ -327,7 +327,17 @@ def findOutputs(collections):
                 if self.clobberOutputs:
                     # only prune
                     _LOG.info("Removing partial outputs for task %s: %s", taskDef, existingRefs)
-                    butler.pruneDatasets(existingRefs, disassociate=True, unstore=True, purge=True)
+                    # Do not purge registry records if this looks like
+                    # an execution butler. This ensures that the UUID
+                    # of the dataset doesn't change.
+                    if butler._allow_put_of_predefined_dataset:
+                        purge = False
+                        disassociate = False
+                    else:
+                        purge = True
+                        disassociate = True
+                    purge = False if butler._allow_put_of_predefined_dataset else True
```

Comment:
Wait, isn't this redundant with the above?

Reply:
Yes. When testing I realized I needed to set the disassociate flag and left this line in by mistake.